pkgs/by-name/fv/fvm/package.nix

```diff
@@ -0,0 +1,49 @@
+{
+  buildDartApplication,
+  installShellFiles,
+  clang,
+  pkg-config,
+  gtk3,
+  cmake,
+  fetchFromGitHub,
+  lib,
+}:
+
+buildDartApplication rec {
+  pname = "fvm";
+  version = "3.2.1";
+  src = fetchFromGitHub {
+    owner = "leoafarias";
+    repo = pname;
+    rev = "refs/tags/${version}";
+    hash = "sha256-i7sJRBrS5qyW8uGlx+zg+wDxsxgmolTMcikHyOzv3Bs=";
+  };
+  autoPubspecLock = "${src}/pubspec.lock";
```

Comment:
This, AFAIK, is not allowed. That's called import from derivation and is basically when the eval step depends on a build step and this is not allowed in nixpkgs.

CI is also complaining that the file is not there.

IDK how it is the state now but in packages I mantained previously I had to save the lockfile converted to json along with the package.nix.

Reply:
Why I cannot? I wanted to use the lock file from the repo to be up-to-date.

Reply:
Nixpkgs assumes one would evaluate all nix code, generate all the derivations then start building.

With this, the eval phase would depend on a build and this is forbidden, and make stuff much slower.